src/server/game/Entities/Object/Object.cpp

```diff
@@ -1213,11 +1213,28 @@ bool WorldObject::IsWithinDist2d(const Position* pos, float dist) const
     return IsInDist2d(pos, dist + GetCombatReach());
 }
 
+bool WorldObject::IsWithinDist2dCenter(float x, float y, float dist) const
```

Comment:
Yea, that's reasonable. I only added it to fit the format of the currently used IsWithinDist2d. If you're ok with directly using that method that's fine with me. I will make that change.